### ISSUE TITLE
storeDir directive does not work with S3

### ISSUE THREAD

**Issue by dzerbino** [nextflow-io/nextflow#2141]
## Bug report 

I have a toy pipeline which looks like this: 

```
process my_process {
    publishDir "s3://dz6-test/"

    input:
    val(param) from Channel.from(1)

    output:
    file("HELLO.tsv") into output_ch

    script:
    """
    echo "Hello, world" > HELLO.tsv
    """
}
```

This version works. However, if I destroy the resulting `s3://my-bucket/test` directory, replace `publishDir` with `storeDir`, then re-run, the job fails with the mention: 

```
Error executing process > 'my_process (1)'

Caused by:
  Missing required nf-amazon module
```

Below is my `nextflow.config` file:

```
plugins {
  id 'nf-amazon'
}

aws {
<Access details to Sanger Institute S3 server redacted>
}
```

### Expected behavior and actual behavior

Both pipelines, with publish- or storeDir should work equally.

### Steps to reproduce the problem

Simply run the pipeline above

### Program output 

```
$ s3cmd del -rf s3://dz6-test/
WARNING: Remote list is empty.
$ ./nextflow run pipeline-publish.nf 
N E X T F L O W  ~  version 21.04.1
Launching `pipeline-publish.nf` [sick_sanger] - revision: 6802ec1d43
executor >  local (1)
[e1/6df91f] process > my_process (1) [100%] 1 of 1 ✔

$ s3cmd del -rf s3://dz6-test/
delete: 's3://dz6-test//'
delete: 's3://dz6-test/HELLO.tsv'
$ ./nextflow run pipeline-store.nf 
N E X T F L O W  ~  version 21.04.1
Launching `pipeline-store.nf` [berserk_fermat] - revision: 5897bc7090
[d3/eca787] process > my_process (1) [100%] 1 of 1, failed: 1
Error executing process > 'my_process (1)'

Caused by:
  Missing required nf-amazon module
```

### Environment 

* Nextflow version: 21.04.1 build 5556
* Java version: 11.0.11
* Operating system: Ubuntu 18.04.4 LTS
* Bash version: 4.4.20(1)-release (x86_64-pc-linux-gnu)

### Additional context

(Add any other context about the problem here)

**Comment by pditommaso** [comment]
Thanks for reporting. Unfortunately, there isn't a quick patch for this. I'd suggest using version 20.10.0 as a temporary workaround

**Comment by pditommaso** [comment]
Thanks for reporting this solved. it's solved on master. It will be included in the next release.

**Comment by pditommaso** [comment]
Included in release `21.06.0-edge`.
